cmake/gdk_toolchain.cmake

```diff
@@ -0,0 +1,92 @@
+# Options to configure the toolchain.
+set(GDK_EDITION 210602 CACHE STRING "GDK edition.")
+set(GDK_PLATFORM Scarlett CACHE STRING "GDK target platform.")
+
+# Required to propagate variables when CMake calls try_compile() to test the toolchain.
+# https://cmake.org/cmake/help/latest/variable/CMAKE_TRY_COMPILE_PLATFORM_VARIABLES.html
+set(CMAKE_TRY_COMPILE_PLATFORM_VARIABLES GDK_EDITION GDK_PLATFORM)
+
+cmake_path(SET gdk_gxdk_path $ENV{GameDK}/${GDK_EDITION}/GXDK NORMALIZE)
+
+# Set C/C++ compile flags and additional include directories.
+foreach(lang C CXX)
+    set(CMAKE_${lang}_FLAGS_INIT "")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /D_GAMING_XBOX")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /DWINAPI_FAMILY=WINAPI_FAMILY_GAMES")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /D_ATL_NO_DEFAULT_LIBS")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /D__WRL_NO_DEFAULT_LIB__")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /D__WRL_CLASSIC_COM_STRICT__")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /D_CRT_USE_WINAPI_PARTITION_APP")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /DWIN32_LEAN_AND_MEAN")
+    string(APPEND CMAKE_${lang}_FLAGS_INIT " /favor:AMD64")
+
+    if(GDK_PLATFORM STREQUAL Scarlett)
+        string(APPEND CMAKE_${lang}_FLAGS_INIT " /D_GAMING_XBOX_SCARLETT")
+        string(APPEND CMAKE_${lang}_FLAGS_INIT " /arch:AVX2")
+    elseif(GDK_PLATFORM STREQUAL XboxOne)
+        string(APPEND CMAKE_${lang}_FLAGS_INIT " /D_GAMING_XBOX_XBOXONE")
+        string(APPEND CMAKE_${lang}_FLAGS_INIT " /arch:AVX")
+    endif()
+
+    set(CMAKE_${lang}_STANDARD_INCLUDE_DIRECTORIES ${gdk_gxdk_path}/gameKit/Include/${GDK_PLATFORM})
+
+    set(CMAKE_${lang}_STANDARD_LIBRARIES "onecoreuap_apiset.lib" CACHE STRING "" FORCE)
+endforeach()
+
+# It's best to avoid inadvertently linking with any libraries not present in the OS.
+set(nodefault_libs "")
+list(APPEND nodefault_libs advapi32.lib)
```

Comment:
Can we just say:

```
list(APPEND nodefault_libs
    advapi32.lib
    comctl32.lib
    comsupp.lib
    ...
)
```

Seems APPEND can take multiple elements:

`list(APPEND <list> [<element> ...])`
https://cmake.org/cmake/help/latest/command/list.html